selftests/functional/runner_vmimage.py

```diff
@@ -0,0 +1,211 @@
+                            if "Fedora 41 x86_64 image path:" in log_content:
+                                self.assertIn(
+                                    "Fedora 41 x86_64 image path:", log_content
+                                )
+                            if "Fedora 41 s390x image path:" in log_content:
+                                self.assertIn(
+                                    "Fedora 41 s390x image path:", log_content
+                                )
+                            if "CentOS 8 x86_64 image path:" in log_content:
+                                self.assertIn(
+                                    "CentOS 8 x86_64 image path:", log_content
+                                )
+                            if "Debian Bullseye (11) x86_64 image path:" in log_content:
+                                self.assertIn(
+                                    "Debian Bullseye (11) x86_64 image path:",
+                                    log_content,
+                                )
+                            if "Ubuntu 22.04 x86_64 image path:" in log_content:
+                                self.assertIn(
+                                    "Ubuntu 22.04 x86_64 image path:", log_content
+                                )
+                            break
+
+            if found_logs:
+                break
+
+        self.assertTrue(
+            found_logs, "Could not find test logs with image path information"
+        )
+
+    @unittest.skipUnless(
+        os.environ.get("AVOCADO_SELFTESTS_NETWORK_ENABLED", False),
+        "Network required to run these tests",
+    )
+    def test_vmimage_runner(self):
+        res = process.run(
+            f"{RUNNER} task-run -i vmimage_0 -k vmimage "
+            "provider=Fedora version=41 arch=x86_64",
+            ignore_status=True,
+            shell=True,
+        )
+        self.assertIn(b"'status': 'started'", res.stdout)
+        self.assertIn(b"'status': 'finished'", res.stdout)
+        self.assertIn(b"'result': 'pass'", res.stdout)
+        self.assertEqual(res.exit_status, 0)
+
+    @unittest.skipUnless(
+        os.environ.get("AVOCADO_SELFTESTS_NETWORK_ENABLED", False),
+        "Network required to run these tests",
+    )
+    def test_vmimage_task(self):
+        res = process.run(
+            f"{RUNNER} task-run -i vmimage_2 -k vmimage "
+            "provider=Fedora version=41 arch=x86_64",
+            ignore_status=True,
+            shell=True,
+        )
+        self.assertIn(b"'status': 'finished'", res.stdout)
+        self.assertIn(b"'result': 'pass'", res.stdout)
+        self.assertIn(b"'id': 'vmimage_2'", res.stdout)
+        self.assertEqual(res.exit_status, 0)
+
+    @unittest.skipUnless(
+        os.environ.get("AVOCADO_SELFTESTS_NETWORK_ENABLED", False),
+        "Network required to run these tests",
+    )
+    def test_vmimage_basic(self):
```

Comment:
IIUIC this is more unit test than functional test, therefore I would have moved it `selftests/functional` dir.